cmd/ingress-shim/controller/sync.go

```diff
@@ -182,7 +185,17 @@ func (c *Controller) setIssuerSpecificConfig(crt *v1alpha1.Certificate, issuer v
 		}
 		switch challengeType {
 		case "http01":
-			domainCfg.HTTP01 = &v1alpha1.ACMECertificateHTTP01Config{Ingress: ing.Name}
+			editInPlace, ok := ingAnnotations[editInPlaceAnnotation]
+			// If annotation isn't present, or it's set to true, edit the existing ingress
+			if ok && editInPlace == "true" {
+				domainCfg.HTTP01 = &v1alpha1.ACMECertificateHTTP01Config{Ingress: ing.Name}
+			} else {
+				ingressClass, ok := ingAnnotations["kubernetes.io/ingress.class"]
+				if !ok {
+					return fmt.Errorf("unable to determine class of ingress %s", ing.Name)
```

Comment:
I'm not sure we should error here - according to the ingress spec, it *is* valid to have a ingress without a class. In this instance, *all* ingress controllers will act on the ingress resource.

Right now, if a user does not specify `ingress` or `ingressClass` on the Certificate resource, an ingress will be created that does not specify ingress class. We should probably do the same here too (i.e. if edit-in-place is not specified, and ingress class isn't specified, just leave the ingress/ingressClass fields blank)